uniffi_bindgen/src/interface/mod.rs

```diff
@@ -216,75 +216,63 @@ impl<'ci> ComponentInterface {
         self.types.get_type_definition(name)
     }
 
-    /// Check whether the given type contains any (possibly nested) Type::Object references.
+    /// Iterate over all the types found within in the given item.
+    /// Unlike the `IterTypes::iter_types` method, this implementation is able to
+    /// recurse into the definitions of user-defined types like records and enums.
+    fn iter_types_in<'a, T: IterTypes>(&'a self, item: &'a T) -> TypeIterator<'a> {
```

Comment:
This is now the single generic place where we do the "recurse through all the types in a given item", and the various other methods and thin wrappers on top of it.